### PR TITLE
MEN-4246: New delta update test to reveal U-Boot integ bug

### PR DESCRIPTION
**Pull request by lluiscampos**
See MEN-4246: New delta update test to reveal U-Boot integ bug

### CODE REVIEW

**Comment by lluiscampos** [comment]
Performing first a full run to verify that the test can reveal the bug. Last commit (the revert of the skip) will be removed before merge.

**Comment by lluiscampos** [comment]
Testing with https://github.com/mendersoftware/mender-qa/pull/469 @ [Pipeline 232808681](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/232808681)

**Comment by lluiscampos** [comment]
New build @ [Pipeline 232984782](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/232984782)

File: tests/acceptance/commercial/test_delta_update_module.py
Comment (kacf) [inline tests/acceptance/commercial/test_delta_update_module.py]: Will you make a separate ticket for this?

File: tests/acceptance/commercial/test_delta_update_module.py
Comment (kacf) [inline tests/acceptance/commercial/test_delta_update_module.py:161]: :+1:

**Review by kacf — APPROVED** [review]

File: tests/acceptance/commercial/test_delta_update_module.py
Comment (oleorhagen) [inline tests/acceptance/commercial/test_delta_update_module.py:127]: Might this as well be a pytest marker?

File: tests/acceptance/commercial/test_delta_update_module.py
Comment (oleorhagen) [inline tests/acceptance/commercial/test_delta_update_module.py:219]:
```suggestion
                f"mender-binary-delta-generator -n v2.0-deltafrom-v1.0 {artifact_from} {artifact_to} -o {artifact_delta}"
```

File: tests/acceptance/commercial/test_delta_update_module.py
Comment (oleorhagen) [inline tests/acceptance/commercial/test_delta_update_module.py:20]: While you're at it, could you sort these?

**Review by oleorhagen — COMMENTED** [review]
Looks pretty

File: tests/acceptance/commercial/test_delta_update_module.py
Comment (lluiscampos) [reply tests/acceptance/commercial/test_delta_update_module.py:127]: Good idea, but maybe in a later PR :)

This was already present for another test, so I just copy-pasted for now.

**Comment by lluiscampos** [comment]
Building with Mender 2.5.x @ [Pipeline 233278637](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/233278637).

Using:
```
release_tool -b 2.5.x --pr meta-mender/1247 --pr mender-qa/469 --pr mender/2.4.1
```

**Comment by lluiscampos** [comment]
Trying also with master @ [Pipeline 233347763](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/233347763).

Using:
```
release_tool -b master --pr meta-mender/1247 --pr mender-qa/469 --pr mender/2.4.1 --pr mender-artifact/3.4.0
```

**Comment by lluiscampos** [comment]
@kacf This is very strange and I cannot find the root cause...

When setting `PREFERRED_VERSION_pn-mender-client = "2.4.1"`, the data partition image is not created properly. In the CI pipelines that I have launched, all tests that use `/data` for something have failed. I can reproduce locally as well. Any idea?

Whatever the problem is, is in `master` and obviously not related to my PR. But in the context of my PR is just another obstacle to actually run the test :sob:

**Comment by kacf** [comment]
Wow, you are really unlucky! :-O

Sorry, no, I cannot say I have any explanation for it. Are you sure it's not the `/var/lib/mender` link?

**Comment by lluiscampos** [comment]
> Are you sure it's not the `/var/lib/mender` link?

Exactly, that was the problem! But not obvious at fist glance how the mistake was introduced... See the last commit in this PR for the fix.

**Comment by lluiscampos** [comment]
> Trying also with master @ [Pipeline 233347763](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/233347763).
> 
> Using:
> 
> ```
> release_tool -b master --pr meta-mender/1247 --pr mender-qa/469 --pr mender/2.4.1 --pr mender-artifact/3.4.0
> ```

New build @ [Pipeline 234816216](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/234816216)

**Comment by lluiscampos** [comment]
> > Trying also with master @ [Pipeline 233347763](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/233347763).
> > Using:
> > ```
> > release_tool -b master --pr meta-mender/1247 --pr mender-qa/469 --pr mender/2.4.1 --pr mender-artifact/3.4.0
> > ```
> 
> New build @ [Pipeline 234816216](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/234816216)

New build @ [Pipeline 234967091](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/234967091)

**Comment by lluiscampos** [comment]
@kacf I run into yet another problem: `mender-binary-delta` cannot run in ARM. The test fails with:
```
time="2020-12-28T11:54:31Z" level=error msg="Module could not be executed: fork/exec /usr/share/mender/modules/v3/mender-binary-delta: no such file or directory"
```

The binary is there, but it expects the linker to be in `lib/ld-linux-armhf.so.3`, which is not present in Yocto. I reported [MEN-4297](https://tracker.mender.io/browse/MEN-4297).

**Comment by lluiscampos** [comment]
The new test showed extremely difficult to fully run in CI due to the different challenges found in the way. The [last run](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/234967091) however at least showed that the test successfully passes on x86 QEMU, so I won't block this PR any longer.

I am now force-pushing with a rebase on master + removing the revert commit (iow skipping the test). The different JIRA tasks are updated for further reference.

File: tests/acceptance/commercial/test_delta_update_module.py
Comment (lluiscampos) [reply tests/acceptance/commercial/test_delta_update_module.py]: Done.

File: meta-mender-core/recipes-mender/mender-client/mender-client_2.3.1.bb
Comment (kacf) [inline meta-mender-core/recipes-mender/mender-client/mender-client_2.3.1.bb:78]: > Exactly, that was the problem! But not obvious at fist glance how the mistake was introduced... See the last commit in this PR for the fix.

Aha! Well spotted!

**Pull request merged**
